vcap/vcap/deprecation.py

```diff
@@ -0,0 +1,41 @@
+import warnings
+from typing import Optional
+from pkg_resources import parse_version
+
+import functools
+
+from . import __version__
+
+
+def deprecated(message: str = "",
+               remove_in: Optional[str] = None,
+               current_version=__version__):
+    """Mark a function as deprecated
+    :param message: Extra details to be added to the warning. For example,
+    the details may point users to a replacement method.
+    :param remove_in: The version when the decorated method will be removed.
+    The default is None, specifying that the function is not currently planned
+    to be removed.
+    :param current_version: Defaults to the vcaps current version.
+    """
+
+    def wrapper(function):
+        warning_msg = f"'{function.__qualname__}' is deprecated "
+        warning_msg += (f"and is scheduled to be removed in {remove_in}. "
+                        if remove_in is not None else ". ")
+        warning_msg += str(message)
+
+        # Only show the DeprecationWarning on the first call
+        warnings.simplefilter("once", DeprecationWarning)
```

Comment:
That's cool

Reply:
Yeah! If I do "once" it only shows once! :tada: